site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1407,6 +1407,102 @@ public function getLateDayInformation($user_id) {
         return $this->course_db->rows();
     }
 
+    public function getLateDayCache() {
+        $query = "SELECT * FROM 
+                    late_day_cache AS ldc
+                    LEFT JOIN gradeable g ON g.g_id=ldc.g_id
+                  ORDER BY late_day_date NULLS LAST, g.g_id NULLS FIRST";
+        $this->course_db->query($query);
+        $return = [];
+
+        // key: user, value: array of all cache available
+        foreach ($this->course_db->rows() as $row) {
+            $user_id = $row['user_id'] ?? $row['team_id'];
+            // title = g_title or event date
+            $title = $row['g_title'] !== null ? $row['g_title'] : explode(" ", $row['late_day_date'])[0];
+            $return[$user_id][$title] = $row;
+        }
+        return $return;
+    }
+
+    public function getLateDayCacheForUser($user_id) {
+        $params = [$user_id];
+        $query = "SELECT * FROM late_day_cache
+                    WHERE user_id=?
+                    ORDER BY late_day_date NULLS LAST, g_id NULLS FIRST";
+        $this->course_db->query($query, $params);
+
+        $index = 1;
+        $late_day_events = [];
+        foreach ($this->course_db->rows() as $row) {
+            // Change string date to DateTime object
+            $row['late_day_date'] = new \DateTime($row['late_day_date']);
+
+            if (isset($row['g_id'])) { // Gradeable late day event
+                $late_day_events[$row['g_id']] = $row;
+            }
+            else { // Late day update event
+                $late_day_events[$index++] = $row;
```

Comment:
Is the intention here to use pre- or post-increment for `$index` here?

Reply:
Hmm, if I'm understanding what this code does properly, it seems *very* prone to someone making changes in the future and accidentally breaking things.  I would recommend doing something like the following just for clarity.
```
$late_day_events[$index] = $row;
$index++; 
```